metals/src/main/scala/scala/meta/internal/metals/ExcludedPackagesHandler.scala

```diff
@@ -0,0 +1,58 @@
+package scala.meta.internal.metals
+
+import scala.collection.mutable.ListBuffer
+
+/**
+ * This class handles packages that should be excluded from indexing causing
+ * them to not be recommended for completions, symbol searches, and code actions.
+ *
+ * @param userConfig access to user input for excluded packages
+ */
+class ExcludedPackagesHandler(userConfig: () => UserConfiguration) {
```

Comment:
Thanks a lot for the review! I went ahead and took some of these suggestions because yea, it simplifies it for sure. I did still incorporate a cache though. Let me know what you think!